### PR TITLE
Remove `Strategy.init_optimizers`

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
After #11155, we can remove `Strategy.init_optimizers` and consolidate in `Strategy.setup_optimizers`.
 
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/optimizer.py [hunk 1]
```diff
@@ -255,9 +255,8 @@ def _configure_schedulers(
     """Convert each scheduler into dict structure with relevant information."""
     lr_schedulers = []
     default_config = _get_default_scheduler_config()
-    # TODO: move is_manual_optimization check out of for loop
-    for scheduler in schedulers:
-        if is_manual_optimization:
+    if is_manual_optimization:
```
Comment (ananthsub) [inline pytorch_lightning/core/optimizer.py:258]: i'd add 2 helper methods for the validation, one for automatic opt and one for manual opt, since they have different conditions/checks

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/core/optimizer.py [hunk 2]
```diff
@@ -249,78 +252,79 @@ def _configure_optimizers(
...
 
-def _configure_schedulers(
-    schedulers: list, monitor: Optional[str], is_manual_optimization: bool
-) -> List[Dict[str, Any]]:
-    """Convert each scheduler into dict structure with relevant information."""
+def _configure_schedulers_automatic_opt(schedulers: list, monitor: Optional[str]) -> List[Dict[str, Any]]:
+    """Convert each scheduler into dict structure with relevant information, when using automatic optimization."""
     lr_schedulers = []
     default_config = _get_default_scheduler_config()
-    # TODO: move is_manual_optimization check out of for loop
```
Comment (ananthsub) [inline pytorch_lightning/core/optimizer.py:258]: this should be a separate PR to ensure this one only does one thing

**Comment by Borda** [comment]
Could we pls use the checklist in the PR template? :rabbit:

File: pytorch_lightning/core/optimizer.py [hunk 3]
```diff
@@ -180,7 +180,10 @@ def _init_optimizers_and_lr_schedulers(model: "pl.LightningModule") -> Tuple[Lis
         optim_conf = _MockOptimizer()
 
     optimizers, lr_schedulers, optimizer_frequencies, monitor = _configure_optimizers(optim_conf)
-    lr_schedulers = _configure_schedulers(lr_schedulers, monitor, not model.automatic_optimization)
+    if model.automatic_optimization:
+        lr_schedulers = _configure_schedulers_automatic_opt(lr_schedulers, monitor)
+    else:
+        lr_schedulers = _configure_schedulers_manual_opt(lr_schedulers, monitor)
```
Comment (Borda) [inline pytorch_lightning/core/optimizer.py:186]:
```suggestion
    _configure = _configure_schedulers_automatic_opt if model.automatic_optimization else _configure_schedulers_manual_opt
    lr_schedulers = _configure(lr_schedulers, monitor)
```
just fancier but not really needed :]

File: pytorch_lightning/core/optimizer.py [hunk 4]
```diff
@@ -249,78 +252,79 @@ def _configure_optimizers(
...
+                )
+            if "interval" in scheduler and scheduler["interval"] not in ("step", "epoch"):
+                raise MisconfigurationException(
+                    'The "interval" key in lr scheduler dict must be "step" or "epoch"'
+                    f' but is "{scheduler["interval"]}"'
+                )
+            scheduler["reduce_on_plateau"] = isinstance(scheduler["scheduler"], optim.lr_scheduler.ReduceLROnPlateau)
+            if scheduler["reduce_on_plateau"] and scheduler.get("monitor", None) is None:
+                raise MisconfigurationException(
+                    "The lr scheduler dict must include a monitor when a `ReduceLROnPlateau` scheduler is used."
```
Comment (Borda) [inline pytorch_lightning/core/optimizer.py:279]:
```suggestion
                    "The LR scheduler `dict` must include a monitor when a `ReduceLROnPlateau` scheduler is used."
```

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/core/optimizer.py (hunk 4 above)
Comment (awaelchli) [reply pytorch_lightning/core/optimizer.py:279]: moved code can't be changed, out of scope.

File: pytorch_lightning/core/optimizer.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/core/optimizer.py:258]: I agree

Comment (daniellepintz) [reply pytorch_lightning/core/optimizer.py]: Done

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/core/optimizer.py [hunk 5]
```diff
@@ -78,12 +78,10 @@ def _to_lightning_optimizer(cls, optimizer: Optimizer, trainer: "pl.Trainer", op
     def toggle_model(self, sync_grad: bool = True) -> Generator[None, None, None]:
         """This function is just a helper for advanced users.
 
-        Considering the current optimizer as A and all other optimizers as B.
-        Toggling means all parameters from B exclusive to A will have ``requires_grad`` set to False.
-
-        When performing gradient accumulation, there is no need to perform grad synchronization
-        during the accumulation phase.
-        Setting `sync_grad` to False will block this synchronization and improve performance.
+        Considering the current optimizer as A and all other optimizers as B. Toggling means all parameters from B
```
[the remark this answers is not in the document]
Comment (daniellepintz) [reply pytorch_lightning/core/optimizer.py:81]: I dont know, working on fixing it

[the remark this answers is not in the document]
Comment (ananthsub) [reply pytorch_lightning/core/optimizer.py]: please update the changelog too

**Comment by daniellepintz** [comment]
Ready to be merged!
